tools/UpdateDotnetRuntime.ps1

```diff
@@ -127,49 +129,103 @@ function Update-CsprojFile([string] $path, $values) {
+            $shouldUpdate = $true
+            $newVersion = $latestSDKversion
+        } else {
+            $shouldUpdate = $false
+            $newVersion = $null
+        }
+    } catch {
+        Write-Verbose -Verbose "Error occured: $_.message"
+        $shouldUpdate = $false
+        $newVersion = $null
+        Write-Error "Error while checking .NET SDK update: $($_.message)"
+    }
 
-# Channel is like: $Channel = "5.0.1xx-preview2"
-$Channel = $dotnetMetadataJson.sdk.channel
+    return @{
+        ShouldUpdate = $shouldUpdate
+        NewVersion   = $newVersion
+        Message      = $Message
+    }
+}
 
-Import-Module "$PSScriptRoot/../build.psm1" -Force
+$dotnetUpdate = Get-DotnetUpdate
 
-Find-Dotnet
+if ($dotnetUpdate.ShouldUpdate) {
 
-if(-not (Get-PackageSource -Name 'dotnet5' -ErrorAction SilentlyContinue))
-{
-    $nugetFeed = ([xml](Get-Content .\nuget.config -Raw)).Configuration.packagesources.add | Where-Object { $_.Key -eq 'dotnet5' } | Select-Object -ExpandProperty Value
-    Register-PackageSource -Name 'dotnet5' -Location $nugetFeed -ProviderName NuGet
-    Write-Verbose -Message "Register new package source 'dotnet5'" -Verbose
-}
+    $dotnetMetadataPath = "$PSScriptRoot/../DotnetRuntimeMetadata.json"
+    $dotnetMetadataJson = Get-Content $dotnetMetadataPath -Raw | ConvertFrom-Json
 
-## Install latest version from the channel
+    # Channel is like: $Channel = "5.0.1xx-preview2"
+    $Channel = $dotnetMetadataJson.sdk.channel
 
-$sdkVersion = if ($SDKVersionOverride) { $SDKVersionOverride } else { "latest" }
+    Import-Module "$PSScriptRoot/../build.psm1" -Force
 
-Install-Dotnet -Channel "$Channel" -Version $sdkVersion
+    Find-Dotnet
 
-Write-Verbose -Message "Installing .NET SDK completed." -Verbose
+    if (-not (Get-PackageSource -Name 'dotnet5' -ErrorAction SilentlyContinue)) {
+        $nugetFeed = ([xml](Get-Content .\nuget.config -Raw)).Configuration.packagesources.add | Where-Object { $_.Key -eq 'dotnet5' } | Select-Object -ExpandProperty Value
+        Register-PackageSource -Name 'dotnet5' -Location $nugetFeed -ProviderName NuGet
+        Write-Verbose -Message "Register new package source 'dotnet5'" -verbose
+    }
 
-$isWindowsEnv = [System.Environment]::OSVersion.Platform -eq [System.PlatformID]::Win32NT
+    ## Install latest version from the channel
 
-$dotnetPath = if ($IsWindowsEnv) { "$env:LocalAppData\Microsoft\dotnet" } else { "$env:HOME/.dotnet" }
+    $sdkVersion = if ($SDKVersionOverride) { $SDKVersionOverride } else { $dotnetUpdate.NewVersion }
 
-$pathSep = [System.IO.Path]::PathSeparator
+    Install-Dotnet -Channel "$Channel" -Version $sdkVersion
 
-if (-not (($ENV:PATH -split $pathSep) -contains "$dotnetPath")) {
-    $env:PATH = "$dotnetPath" + $pathSep + "$ENV:PATH"
-}
+    Write-Verbose -Message "Installing .NET SDK completed." -Verbose
+
+    $isWindowsEnv = [System.Environment]::OSVersion.Platform -eq [System.PlatformID]::Win32NT
```

Comment:
should we be using the environment from build.psm1?

Reply:
We might as well. Did not know we had a `Get-EnvironmentInformation` function, I thought it was just $environment.

Reply:
Fixed